### PR TITLE
chore: write tasks for OH

### PR DESCRIPTION
**Pull request by lazaralex98**
Every time I make edits to the OffsetHelper I need to re-deploy and re-verify. I coded a task to make this process easier.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
         env:
           POLYGON_URL: ${{ secrets.POLYGON_URL }}
 
-      - name: Deploy contract
-        run: yarn hardhat --network hardhat run ./scripts/deployOffsetHelper.ts
-        env:
-          POLYGON_URL: ${{ secrets.POLYGON_URL }}
+      # - name: Deploy contract
+      #   run: yarn hardhat --network hardhat deployOffsetHelper
```
Comment (kargakis) [inline .github/workflows/ci.yml:65]: We could have a `--verify` flag provided in the `deployOffsetHelper` task to run the verification conditionally and set it here to `false`.

Comment (lazaralex98) [reply .github/workflows/ci.yml:65]: Makes sense. I'll amend the above commit

File: hardhat.config.ts [hunk 2]
```diff
@@ -17,6 +20,46 @@ task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
   }
 });
 
+task("deployOffsetHelper", "Deploys and verifies OffsetHelper")
+  .addFlag("verify", "Set true to verify the OffsetHelper after deployment")
```
Comment (kargakis) [inline hardhat.config.ts:24]: Let's set this by default to `true` so we won't have to remember to set the flag when running locally and explicitly set to `false` in CI.
```suggestion
  .addFlag("verify", "Set true to verify the OffsetHelper after deployment", true)
```

Comment (lazaralex98) [reply hardhat.config.ts:24]: Wouldn't it make more sense to have a `dontVerify` flag instead in this case?

Comment (lazaralex98) [reply hardhat.config.ts:24]: Also, if we commit this the CI will fail because it'll try to verify

**Comment by lazaralex98** [comment]
Should be good now

**Review by kargakis — APPROVED** [review]

**Comment by lazaralex98** [comment]
Is it normal for the deploy task to take 10m @kargakis ?

File: hardhat.config.ts [hunk 3]
```diff
@@ -17,6 +21,51 @@ task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
...
+      addressesToUse.nct,
+      addressesToUse.usdc,
+      addressesToUse.weth,
+      addressesToUse.wmatic,
+    ]);
+    await oh.deployed();
+    console.log(`OffsetHelper deployed on ${hre.network.name} to:`, oh.address);
+
+    if (taskArgs.verify === true) {
+      await oh.deployTransaction.wait(5);
```
Comment (kargakis) [inline hardhat.config.ts:48]: Can we just remove this? `deployed()` should work for the most part, no?

Comment (kargakis) [reply hardhat.config.ts:48]: I guess there is still a race between the node we use and the node used by Polygonscan... Is `5` here 5 confirmations?

Comment (lazaralex98) [reply hardhat.config.ts:48]: Nope we can't. It takes a while for Polygonscan to index the deployment transaction. Hardhat docs recommend we wait for 5 confirmations in automated tasks to make sure we can actually run the verifications.

**Pull request merged**